### PR TITLE
Added exempting variable annotations

### PR DESCRIPTION
**Pull request by sergeykad**
Partially fixes https://github.com/google/error-prone/issues/2881

### CODE REVIEW

File: core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java [hunk 1]
```diff
@@ -136,7 +136,10 @@
           "org.openqa.selenium.support.FindBy",
           "org.openqa.selenium.support.FindBys",
           "org.apache.beam.sdk.transforms.DoFn.TimerId",
-          "org.apache.beam.sdk.transforms.DoFn.StateId");
+          "org.apache.beam.sdk.transforms.DoFn.StateId",
+          "org.mockito.Mock",
```
Comment (cushon) [inline core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java:140]: Why should `@Mock` variables be exempted, isn't it generally unnecessary to have unused mocks?

**Review by cushon — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java (hunk 1 above)
Comment (sergeykad) [reply core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java:140]: It can be used indirectly, for example by `@InjectMocks` annotation.

Comment (cushon) [reply core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java:140]: What do you think about adding `@InjectMocks` to the list instead? I'd like to keep the ability to recognize unused fields that are directly annotated with `@Mock`

Comment (sergeykad) [reply core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java:140]: `@InjectMocks` is added to another field, so it does not really help. Only Mockito itself can know for sure if it is used and complain about it, which it does.

If you are feeling that it is not good enough I will remove the `@Mock` annotation altogether.

Comment (sergeykad) [reply core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java:140]: So, what should I do?

Comment (cushon) [reply core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java:140]: (Sorry for losing track of this.)

I think for the uses of UnusedVariable in my employer's codebase, we'd probably rather have the false positive than the false negatives.

An alternative to enabling `@Mock` by default could be to add a flag (similar to the existing `-XepOpt:Unused:methodAnnotationsExemptingParameters` option supported in this checker) to allow configuring it to skip fields annotated with `@Mock`. What do you think?

**Review by cushon — CHANGES_REQUESTED** [review]

**Pull request closed without merging**